gandalf.js

```diff
@@ -1,121 +1,128 @@
 /**
```

Comment:
This block of code could benefit from some comments explaining the logic behind it.